CMakeLists.txt

```diff
@@ -18,6 +18,13 @@ if(NOT BUILD_SHARED_LIBS)
   set(LOG4CXX_COMPILE_DEFINITIONS LOG4CXX_STATIC)
 endif()
 
+option(EXPORT_ONLY_LOG4CXX_SYMBOLS "Only exports the log4cxx symbols" ON)
+if(BUILD_SHARED_LIBS AND EXPORT_ONLY_LOG4CXX_SYMBOLS)
+  SET(LINKER_VERSION_SCRIPT_FILE "${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/log4cxx.map")
+  SET(LINKER_VERSION_SCRIPT_FLAG "-Wl,--version-script=${LINKER_VERSION_SCRIPT_FILE}")
```

Comment:
Does this work for any compiler supported by CMAKE? The arguments look pretty compiler-specific to me.

Reply:
This is a linker option, supported by GNU ld and GNU gold.

Reply:
So `log4cxx.map` is linker-specific and might be named more specifically? Something like `log4cxx.gnu[_ld].map` or something like that? You already mentioned visual studio, so something like `log4cxx.vs.def` might need to be added in future as well? Do all those files share some common name one could reuse, something about symbols or exporting or linking or ...?